csrc/multidevice/utils.h

```diff
@@ -86,8 +86,22 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
+// Shards all tensors in tvs like reference
 void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs);
 
+// Shards all TVs from the output of from expressions to the outputs of the to
+// expressions like the reference TV.
+void shardBetween(
+    const std::vector<Expr*>& from,
+    const std::vector<Expr*>& to,
+    TensorView* ref);
+
+// Shards all TVs between from_tvs and to_tvs like the reference TV.
```

Comment:
"between" isn't accurate -- as you said in the .cpp comment, the function processes a superset of the between expressions. I think it's worth calling out the behavior in the .h comment instead of .cpp.

Reply:
how about `shardFrom`?

Reply:
I think that's a better name. However, even with that, shardFrom makes me think it's similar to getAllTvsFrom while it's not. In getAllTvsFrom, cutoff_tv_set needs to surround from_tvs, and shardFrom's `to` doesn't need to. I think that behavior is still worth calling out in the header comment.

Reply:
How about `propagateShardings` then? With the updated comment of the scope of TVs that get affected, this will clear up some confusion with `getTvsBetween` and `getAllTvsFrom` behavior?

Reply:
Do you mean renaming this function to `propagateShardings`? That makes sense. I just don't know how to deal with the existing `propagateShardings` :) 

Sorry, I don't want to bikeshed too much on naming. I wasn't opposed to `shardBetween` -- it is probably still the closest to the actual behavior. It was just the fact that the result TVs are not strictly "between" `from` and `to` that made me think some header comments are worthwhile.

Reply:
sure I changed it back to `shardBetween` to avoid confusion with the propagateSharding pass. Comment has been updated as well.